src/index.ts

```diff
@@ -287,308 +563,58 @@ export function buildStorageKeySuffix(apiKey: string): string {
-          // values into the config store (ex:the cache is not expired or invalid config data was received).
-          // It's important not to set an error here as this state does not mean that an error was encountered.
-          // If the initial cache had not expired, the fetch would be skipped and the config store would report as
-          // uninitialized because only a successful fetch setting the data entries will set the persistent store as
-          // initialized.
-          return ConfigLoaderStatus.DID_NOT_PRODUCE;
-        }
-      })
-      .catch((e) => {
-        applicationLogger.warn('Eppo SDK encountered an error initializing from fetching', e);
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        initFromFetchError = e;
-        return ConfigLoaderStatus.FAILED;
-      })
-      .then((status) => {
-        return { source: ConfigSource.FETCH, result: status };
-      });
-
-    const racers: ConfigurationLoadAttempt[] = [attemptInitFromConfigStore];
-
-    // attemptInitFromFetch will return early and not attempt to fetch if `skipInitialRequest` is `true`.
-    // We could still race it against a load from configuration as the handling logic will see that no config was loaded
-    // into the config store from the remote fetch and return `DID_NOT_PRODUCE`.
-    if (!config.skipInitialRequest) {
-      racers.push(attemptInitFromFetch);
-    }
-    let initializationSource: ConfigSource;
-    const { source: firstSource, result: firstConfigResult } = await Promise.race(racers);
+  const instance = getInstance();
 
-    if (firstConfigResult === ConfigLoaderStatus.COMPLETED) {
-      // First config load successfully produced a configuration
-      initializationSource = firstSource;
+  if (EppoJSClient.initialized) {
+    if (config.forceReinitialize) {
+      applicationLogger.info(
+        'Eppo SDK is already initialized, reinitializing since forceReinitialize is true.',
+      );
     } else {
-      // First attempt failed or did not produce configuration, but we have a second at bat that will be executed in the scope of the top-level try-catch
-
-      const secondAttempt =
-        firstSource === ConfigSource.FETCH ? attemptInitFromConfigStore : attemptInitFromFetch;
-
-      initializationSource = await secondAttempt.then((resultPair: ConfigLoadResult) => {
-        const { source, result } = resultPair;
-        return result === ConfigLoaderStatus.COMPLETED ? source : ConfigSource.NONE;
-      });
-    }
-
-    if (initializationSource === ConfigSource.NONE) {
-      // Neither config loader produced useful config. Return error, if exists, in order from first to last: fetch, config store, generic.
-      initializationError = initFromFetchError
-        ? initFromFetchError
-        : initFromConfigStoreError
-          ? initFromConfigStoreError
-          : new Error('Eppo SDK: No configuration source produced a valid configuration');
+      applicationLogger.warn(
```

Comment:
The diff makes it hard to see what's left in this method;

1) We keep the `forceReinitialize` check here to keep it out of the EppoJSClient.init method
2) Warn if an initialization is currently in progress